subtests/docker_cli/run_flags/run_flags.py

```diff
@@ -0,0 +1,94 @@
+"""
+Summary
+-------
+Checks that docker daemon is running with expected command-line options.
+
+Operational Summary
+-------------------
+Get the full command line of the currently-running docker or docker-latest
+daemon. Check for presence of expected option flags.
+
+"""
+
+import os
+from distutils.version import LooseVersion   # pylint: disable=E0611,F0401
+import dockertest.docker_daemon
+from dockertest import subtest
+from dockertest.output import DockerVersion
+from dockertest.xceptions import DockerTestFail, DockerTestNAError
+
+
+class run_flags(subtest.SubSubtestCaller):
+    pass
+
+
+class run_flags_base(subtest.SubSubtest):
+
+    def _has_storage_opt(self, wanted):
+        docker_cmdline = dockertest.docker_daemon.cmdline()
+        last_opt = ''
+        for opt in docker_cmdline:
+            if opt == wanted and last_opt == '--storage-opt':
+                return True
+            last_opt = opt
+        self.logwarning("Option '%s' not in %s" % (wanted, docker_cmdline))
+        return False
+
+
+class run_flags_deferred_removal(run_flags_base):
+    """
+    deferred REMOVAL should always be set in all docker versions we test.
+    """
+
+    def postprocess(self):
+        super(run_flags_deferred_removal, self).postprocess()
+        if self._has_storage_opt('dm.use_deferred_removal=true'):
+            return
+        raise DockerTestFail("missing option: dm.use_deferred_removal")
+
+
+class run_flags_deferred_deletion(run_flags_base):
+    """
+    deferred DELETION should be set in docker >= 1.12 but only if
+    the kernel supports it.
+    """
+
+    def postprocess(self):
+        super(run_flags_deferred_deletion, self).postprocess()
+
+        # Check flag first. It's quick, and if it passes, we're good.
+        if self._has_storage_opt('dm.use_deferred_deletion=true'):
+            return
+
+        # Flag missing. This is OK under some circumstances.
+        DockerVersion().require_server("1.12")
+        self._check_kernel_support()
+
+        raise DockerTestFail("missing option: dm.use_deferred_deletion")
+
+    @staticmethod
+    def _is_rhel_or_centos():
+        """ True if this system is RHEL or CentOS """
+        try:
+            with open('/etc/redhat-release', 'r') as rh_release_fh:
+                rh_release = rh_release_fh.read()
+            if 'Red Hat Enterprise Linux' in rh_release:
+                return True
+            if 'CentOS' in rh_release:
+                return True
+        except IOError:
+            return False
+        return False
+
+    def _check_kernel_support(self):
+        """
+        Fail with N/A if this is a RHEL/CentOS host with too-old kernel.
+        """
+        if not self._is_rhel_or_centos():
+            return
+        required = LooseVersion("3.999")    # FIXME once we ID a good version
```

Comment:
`is_rhel_or_centos`: removed. Thanks, much cleaner now. I can assume that CentOS will track RHEL, and that Fedora will have much more recent kernels.

Config setting: I'm moderately opposed to that, because I don't see it being configurable. It's a value we'll discover some day. Once we do we'll make a new commit here (have to do that whether it's a config value or in code), and forget about it forever. It should never need to be touched again, which means it shouldn't offer the illusion of configurability. But there are still uncertainties about this, so if you feel strongly about making it configurable I'd like to hear your thoughts.